core/capabilities/ccip/ccipevm/msghasher_test.go

```diff
@@ -328,11 +333,105 @@ func TestMessagerHasher_againstRmnSharedVector(t *testing.T) {
 				TokenAmounts:   tokenAmounts,
 			}
 
-			any2EVMMessage = ccipMsgToAny2EVMMessage(t, msg)
+			any2EVMMessage = ccipMsgToAny2EVMMessage(t, msg, sourceChainSelector)
 		)
 
 		const (
-			rmnMsgHash = "0xb6ea678f918293745bfb8db05d79dcf08986c7da3e302ac5f6782618a6f11967"
+			rmnMsgHash = "0x85f1d7bc457baf3c72e90aced994a501504f1a5e8d21cd80aaaf2e2d66a155d6"
+		)
+
+		h := NewMessageHasherV1(logger.Test(t), ExtraDataCodec)
+		msgH, err := h.Hash(tests.Context(t), msg)
+		require.NoError(t, err)
+
+		msgHashOnchain, err := msghasher.Hash(&bind.CallOpts{
+			Context: tests.Context(t),
+		}, any2EVMMessage, onRampAddress)
+		require.NoError(t, err)
+
+		t.Logf("rmn hash: %s, onchain hash: %s, my hash: %s", rmnMsgHash, hexutil.Encode(msgHashOnchain[:]), msgH.String())
+		require.Equal(t, msgHashOnchain, [32]byte(msgH), "my hash and onchain hash should match")
+		require.Equal(t, rmnMsgHash, msgH.String(), "rmn hash and my hash should match")
+	})
+
+	t.Run("solana vector", func(t *testing.T) {
+		key := solanago.MustPublicKeyFromBase58("C8WSPj3yyus1YN3yNB6YA5zStYtbjQWtpmKadmvyUXq8")
+		require.NoError(t, err)
+
+		// Borsh encoded extra args
+		ea := fee_quoter.EVMExtraArgsV2{
+			GasLimit:                 agbinary.Uint128{Lo: 5000, Hi: 0},
+			AllowOutOfOrderExecution: false,
+		}
+
+		var extraArgsbuf bytes.Buffer
+		encoder := agbinary.NewBorshEncoder(&extraArgsbuf)
+		err = ea.MarshalWithEncoder(encoder)
+		require.NoError(t, err)
+
+		destGasAmount := uint32(10)
+		destExecData := make([]byte, 4)
+		binary.LittleEndian.PutUint32(destExecData, destGasAmount)
+
+		var (
+			// header fields
+			messageID           = mustBytes32FromString(t, "0xcdad95e113e35cf691295c1f42455d41062ba9a1b96a6280c1a5a678ef801721")
+			sourceChainSelector = cciptypes.ChainSelector(124615329519749607)   // solana
+			destChainSelector   = cciptypes.ChainSelector(16015286601757825753) // sepolia
+			sequenceNumber      = cciptypes.SeqNum(386)
+			nonce               = uint64(1)
+			// message fields
+			// sender is parsed unpadded since its emitted unpadded from EVM.
+			senderAddress = cciptypes.UnknownAddress(hexutil.MustDecode("0x269895AC2a2eC6e1Df37F68AcfbBDa53e62b71B1"))
```

Comment:
A lot of these fields are not correctly set in the context of SVM. E.g, the addresses on the source size are hex-encoded. Can we make this a proper SVM message? Otherwise the test is not very useful or indicative.